src/ripple/rpc/impl/TransactionSign.cpp

```diff
@@ -150,25 +150,19 @@ TxnSignApiFacade::findPathsForOneIssuer (
     STPathSet const& paths,
     STPath& fullLiquidityPath) const
 {
-    if (! ledger_) // Unit testing.
-    {
-        // Note that unit tests don't (yet) need paths or fullLiquidityPath.
-        boost::optional<STPathSet> result;
-        result.emplace();
-        return result;
-    }
-
-    auto cache = std::make_shared<RippleLineCache> (ledger_);
-    return ripple::findPathsForOneIssuer(
-        cache,
-        accountID_,
-        dstAccountID,
-        srcIssue,
-        dstAmount,
-        searchLevel,
-        maxPaths,
-        paths,
-        fullLiquidityPath);
+    // Unit testing. Note that unit tests don't need paths or fullLiquidityPath.
+    if (! ledger_)
+        return STPathSet();
+
+    Pathfinder pf(std::make_shared<RippleLineCache>(ledger_),
+        accountID_, dstAccountID, srcIssue.currency,
+            srcIssue.account, dstAmount, boost::none);
+    if (! pf.findPaths(searchLevel))
+        return boost::none;
+
+    pf.computePathRanks(maxPaths);
+    return pf.getBestPaths(maxPaths, fullLiquidityPath,
+        paths, srcIssue.account);
```

Comment:
#1289 gets rid of this class entirely. https://github.com/vinniefalco/rippled/commit/d709ff43dc6fd11565b9a6baa84df846344c70bd